frame/collective/src/lib.rs

```diff
@@ -708,10 +730,128 @@ decl_module! {
+		}]
+		fn dispatch_as_quantity_account(origin,
+			proposal: Box<T::Proposal>,
+			#[compact] length_bound: u32,
+		) -> DispatchResult {
+			let maybe_raw_origin: Result<RawOrigin<T::AccountId, I>, _> = <T as Config<I>>::Origin::from(origin).into();
+
+			let raw_origin = maybe_raw_origin.map_err(|_| DispatchError::BadOrigin)?;
+
+			let proposal_len = proposal.using_encoded(|x| x.len());
+			ensure!(proposal_len <= length_bound as usize, Error::<T, I>::WrongProposalLength);
+
+			let account = Self::origin_to_quantity_account(raw_origin)?;
+			let proposal_hash = T::Hashing::hash_of(&proposal);
+			let signed_origin = frame_system::RawOrigin::Signed(account.clone());
+			let result = proposal.dispatch(signed_origin.into());
+			Self::deposit_event(
+				RawEvent::AccountExecuted(account, proposal_hash, result.map(|_| ()).map_err(|e| e.error))
+			);
+			Ok(())
+		}
 	}
 }
 
 impl<T: Config<I>, I: Instance> Module<T, I> {
+	/// Get the unique index for this pallet from the `construct_runtime` order.
+	pub fn pallet_index() -> u32 {
+		 T::PalletInfo::index::<Pallet<T, I>>().expect("pallet is part of the runtime and thus has an index; qed.") as u32
+	}
+
+	/// A simple helper function to generate a collective ratio account from the collective origin.
+	///
+	/// A ratio account is an account which represents a ratio of n / d users where n is the number
+	/// of approving members, and d is the total number of members. n and d can be any fraction which
+	/// is less or equally restrictive than the origin.
+	pub fn origin_to_ratio_account(
+		origin: RawOrigin<T::AccountId, I>,
+		target_n: u32,
+		target_d: u32,
+	) -> Result<T::AccountId, DispatchError> {
+		ensure!(target_n <= target_d, Error::<T, I>::NotEnoughMembers);
+		match origin {
+			RawOrigin::Members(n, d) => {
+				// n is the number of members who voted aye. d is the total number of members.
+				// Should never be able to make an origin which represents more members than exists.
+				ensure!(target_d <= d, Error::<T, I>::NotEnoughMembers);
+				// The target ratio should be less than or equal to the ratio provided by the origin.
+				// For example 6/13 cannot create a 1/2 account, but 7/13 can.
+				// To check (target_n / target_d) <= (n / d) we check (target_n * d) <= (target_d * n)
+				let target_n_d = target_n.checked_mul(d).ok_or(Error::<T, I>::Overflow)?;
+				let target_d_n = target_d.checked_mul(n).ok_or(Error::<T, I>::Overflow)?;
+				ensure!(target_n_d <= target_d_n, Error::<T, I>::NotEnoughMembers);
+				// Create the account.
+				Ok(PALLET_ID.into_sub_account((Self::pallet_index(), target_n, target_d)))
+			},
+			_ => return Err(DispatchError::BadOrigin.into()),
+		}
+	}
+
+	/// A simple helper function to generate a collective quantity account from the collective origin.
+	///
+	/// A quantity account simply represents the total number of members used to generate this account.
+	pub fn origin_to_quantity_account(
```

Comment:
isn't this a bit intermixed? In my head these two arms are different things. Consider: 

Alice is the only member in the council. `RawOrigin::Members(1, _)` is not the same as `RawOrigin::Member(alice)` whereas it should be? 

I guess the use cases here is for a parachain to say: 

1. allow this call if any 3 council member approve it which is generated via `RawOrigin::Members(n, _)`
2. allow this call if alice approves it, which is generated `RawOrigin::Member(who)`

probably not a big deal but I'd expect to see 3 new dispatches 

1. `dispatch_as_quantity`
2. `dispatch_as_quantity`
3. `dispatch_as_given`

Reply:
It just seemed to be to be one less extrinsic / function needed. Naming can potentially be improved, but i dont see why to increase the footprint when less will do the same.